python/lsst/pex/config/dictField.py

```diff
@@ -164,7 +182,7 @@ def __reduce__(self):
         )
 
 
-class DictField(Field):
+class DictField(Field, Generic[KeyTypeVar, ItemTypeVar]):
```

Comment:
I think this should be `Field[Dict[KeyTypeVar, ItemTypeVar]], Generic[KeyTypeVar, ItemTypeVar]` as the bases.